source/extensions/load_balancing_policies/client_side_weighted_round_robin/client_side_weighted_round_robin_lb.cc

```diff
@@ -146,14 +161,25 @@ void ClientSideWeightedRoundRobinLoadBalancer::updateWeightsOnHosts(const HostVe
   if (!weights.empty()) {
```

Comment:
IIUC there's no need to compute the `default_weight` if all hosts have an assigned weight.
It may be better to put the entire `if` block and the following `for` block inside a `if (!hosts_with_default_weight.empty())` block.